optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/incremental/AbstractIncrementalScoreCalculator.java

```diff
@@ -1,29 +0,0 @@
-/*
- * Copyright 2012 Red Hat, Inc. and/or its affiliates.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.optaplanner.core.impl.score.director.incremental;
-
-import org.optaplanner.core.api.domain.solution.PlanningSolution;
-
-/**
- * Abstract superclass for {@link IncrementalScoreCalculator}.
- *
- * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
- * @see IncrementalScoreCalculator
- */
-public abstract class AbstractIncrementalScoreCalculator<Solution_> implements IncrementalScoreCalculator<Solution_> {
```

Comment:
+1 to remove this, but will need an upgradeREcipe line and a a deprecation in 7.x - just like Radovan did for the AbstractCustomerPhaseCommand